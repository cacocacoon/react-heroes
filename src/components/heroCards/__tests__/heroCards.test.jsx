import React from 'react'
import thunk from 'redux-thunk'
import { MemoryRouter as Router } from 'react-router-dom'
import { cleanup } from '@testing-library/react'
import configureMockStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import TestRenderer, { act } from 'react-test-renderer'

import HeroCards from '../heroCards'

jest.mock('rc-queue-anim', () => {
	return function MockQueueAnim(props) {
		return props.children
	}
})

global.fetch = jest.fn().mockImplementation(() => (
	Promise.resolve({
		json: () => Promise.resolve([])
	})
))

afterEach(cleanup)
const mockStore = configureMockStore([thunk])

function TestHeroCards({ store }) {
	return (
		<Provider store={store} >
			<Router>
				<HeroCards />
			</Router>
		</Provider>
	)
}

describe('heroCards', () => {
	test('empty hero', async () => {
		const store = mockStore({ hero: { list: [] }, ui: { selectedHeroId: 1 } })
		let heroCards
		await act(async () => {
			heroCards = TestRenderer.create(
				<TestHeroCards store={store} />
			)
		})

		expect(heroCards.toJSON()).toMatchSnapshot()
	})

	test('include heros', async () => {
		const store = mockStore({
			hero: {
				list: [{
					id: "1",
					image: "http://i.annihil.us/u/prod/marvel/i/mg/6/90/537ba6d49472b/standard_xlarge.jpg",
					name: "Daredevil"
				}, {
					id: "2",
					image: "http://i.annihil.us/u/prod/marvel/i/mg/6/90/537ba6d49472b/standard_xlarge.jpg",
					name: "Test"
				}]
			},
			ui: { selectedHeroId: 1 }
		})
		let testHeroCards
		await act(async () => {
			testHeroCards = TestRenderer.create(
				<TestHeroCards store={store} />
			)
		})

		expect(testHeroCards).toMatchSnapshot()
	})
})