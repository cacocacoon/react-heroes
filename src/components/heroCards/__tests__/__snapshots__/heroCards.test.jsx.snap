// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`heroCards empty hero 1`] = `
<div
  className="hero-cards-wrapper"
>
  <i
    aria-label="icon: loading"
    className="anticon anticon-loading"
    style={
      Object {
        "fontSize": 24,
      }
    }
  >
    <svg
      aria-hidden="true"
      className="anticon-spin"
      data-icon="loading"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
    >
      <path
        d="M988 548c-19.9 0-36-16.1-36-36 0-59.4-11.6-117-34.6-171.3a440.45 440.45 0 0 0-94.3-139.9 437.71 437.71 0 0 0-139.9-94.3C629 83.6 571.4 72 512 72c-19.9 0-36-16.1-36-36s16.1-36 36-36c69.1 0 136.2 13.5 199.3 40.3C772.3 66 827 103 874 150c47 47 83.9 101.8 109.7 162.7 26.7 63.1 40.2 130.2 40.2 199.3.1 19.9-16 36-35.9 36z"
      />
    </svg>
  </i>
  <div
    className="hero-cards"
  />
</div>
`;

exports[`heroCards include heros 1`] = `
<div
  className="hero-cards-wrapper"
>
  <div
    className="hero-cards"
  >
    <div
      className="card-wrapper"
    >
      <a
        href="/heroes/1"
        onClick={[Function]}
      >
        <div
          className="ant-card custom-card  ant-card-bordered"
        >
          <div
            className="ant-card-body"
            style={Object {}}
          >
            <img
              alt="Daredevil"
              className="hero-img"
              src="http://i.annihil.us/u/prod/marvel/i/mg/6/90/537ba6d49472b/standard_xlarge.jpg"
            />
            <p>
              Daredevil
            </p>
          </div>
        </div>
      </a>
    </div>
    <div
      className="card-wrapper"
    >
      <a
        href="/heroes/2"
        onClick={[Function]}
      >
        <div
          className="ant-card custom-card  ant-card-bordered"
        >
          <div
            className="ant-card-body"
            style={Object {}}
          >
            <img
              alt="Test"
              className="hero-img"
              src="http://i.annihil.us/u/prod/marvel/i/mg/6/90/537ba6d49472b/standard_xlarge.jpg"
            />
            <p>
              Test
            </p>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
